src/jemalloc.c

```diff
@@ -2659,25 +2678,42 @@ je_realloc(void *ptr, size_t arg_size) {
 
 	if (unlikely(size == 0)) {
 		if (ptr != NULL) {
-			/* realloc(ptr, 0) is equivalent to free(ptr). */
```

Comment:
nit: move this to a separate static function?

Also, maybe let's add a counter in the stats? I think a simple global atomic would be fine.

How about we disable tcache by default for the strict case? So that for the leaky behavior, at least it doesnt contribute to fragmentation much.  On the other hand, we can also say we should not try to "hide" the leak though (either way the counter above should still serve as an indicator).